api/explorer/api/ingest.py

```diff
@@ -176,27 +121,46 @@ def ingest_nix_graph(nix_graph: model.NixGraph, g: GraphTraversalSource) -> None
     """
 
     # Note: As an optimization we split logic for writing the graph into two
-    # functions. The first one ensures that every root node of the graph exists
-    def write_root_node_to_graph(root_package: graph.Package) -> None:
-        graph.insert_unique_vertex(root_package, g)
-
-    # Then, this second function handles writing the next level of the graph as well
-    # as the edge connecting the levels, assuming that the current level already exists
-    # in the Gremlin graph.
-    def write_layer_to_graph(
+    # functions. The first one ensures that every node of the graph exists
+    def write_node_to_graph(package: graph.Package) -> None:
+        graph.insert_unique_vertex(package, g)
+
+    # Then, this second function handles writing the edges of the graph
+    def write_edge_to_graph(
         current_package: graph.Package,
-        upstream_package: graph.Package,
+        dependency_package: graph.Package,
         edge: graph.Edge,
     ) -> None:
-        graph.insert_unique_vertex(upstream_package, g)
         graph.insert_unique_directed_edge(
             edge,
             from_vertex=current_package,
-            to_vertex=upstream_package,
+            to_vertex=dependency_package,
             g=g,
         )
 
-    traverse(nix_graph, write_root_node_to_graph, write_layer_to_graph)
+    # Write nodes to the Gremlin graph
+    click.echo("Writing nodes to the Gremlin graph...")
+    with click.progressbar(nix_graph.packages) as bar:
+        for model_pkg in bar:
+            pkgs = safe_parse_package(model_pkg)
+            for pkg in pkgs:
+                write_node_to_graph(pkg)
+
+    # Write edges to the Gremlin graph
+    click.echo("Writing edges to the Gremlin graph...")
+    with click.progressbar(nix_graph.packages) as bar:
+        for model_pkg in bar:
```

Comment:
(nitpick, can be ignored)
```suggestion
    with click.progressbar(nix_graph.packages) as model_pkgs_with_bar:
        for model_pkg in model_pkgs_with_bar:
```